Telerik.JustMock/Expectations/Local/LocalExpectation.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Telerik.JustMock.Core;
+using Telerik.JustMock.Expectations.Abstraction.Local;
+using Telerik.JustMock.Expectations.Abstraction.Local.Function;
+
+namespace Telerik.JustMock.Expectations
+{
+	internal sealed class LocalExpectation : ILocalExpectation
+	{
+		public IFunctionExpectation Function {
+			get
+			{
+				return ProfilerInterceptor.GuardInternal(() => new FunctionExpectation());
```

Comment:
@lyubomir-rusev, do we really need to create a new instance each time, IMO having just one I fine. Also it would be better to move guarding inside IFunctionExpectation method implementations. What do you think?

Reply:
Guarding was added for functions from IFunctionExpectation in latest commits.